pkg/clients/sqs/sqs.go

```diff
@@ -0,0 +1,237 @@
+	UntagQueueRequest(input *sqs.UntagQueueInput) sqs.UntagQueueRequest
+}
+
+// NewClient creates new Queue Client with provided AWS Configurations/Credentials
+func NewClient(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (Client, error) {
+	cfg, err := auth(ctx, credentials, awsclients.DefaultSection, region)
+	if cfg == nil {
+		return nil, err
+	}
+	return sqs.New(*cfg), err
+}
+
+// GenerateCreateAttributes returns a map of queue attributes for Create operation
+func GenerateCreateAttributes(p *v1alpha1.QueueParameters) map[string]string {
+	m := GenerateQueueAttributes(p)
+	if p.FifoQueue != nil {
+		m[v1alpha1.AttributeFifoQueue] = fmt.Sprint(*p.FifoQueue)
+	}
+	return m
+}
+
+// GenerateUpdateAttributes returns a map of queue attributes for update operation
+func GenerateUpdateAttributes(p *v1alpha1.QueueParameters) map[string]string {
+	return GenerateQueueAttributes(p)
+}
+
+// GenerateQueueAttributes returns a map of queue attributes
+func GenerateQueueAttributes(p *v1alpha1.QueueParameters) map[string]string { // nolint:gocyclo
+	m := map[string]string{}
+	if p.DelaySeconds != nil {
+		m[v1alpha1.AttributeDelaySeconds] = strconv.FormatInt(aws.Int64Value(p.DelaySeconds), 10)
+	}
+	if p.MaximumMessageSize != nil {
+		m[v1alpha1.AttributeMaximumMessageSize] = strconv.FormatInt(aws.Int64Value(p.MaximumMessageSize), 10)
+	}
+	if p.MessageRetentionPeriod != nil {
+		m[v1alpha1.AttributeMessageRetentionPeriod] = strconv.FormatInt(aws.Int64Value(p.MessageRetentionPeriod), 10)
+	}
+	if p.ReceiveMessageWaitTimeSeconds != nil {
+		m[v1alpha1.AttributeReceiveMessageWaitTimeSeconds] = strconv.FormatInt(aws.Int64Value(p.ReceiveMessageWaitTimeSeconds), 10)
+	}
+	if p.VisibilityTimeout != nil {
+		m[v1alpha1.AttributeVisibilityTimeout] = strconv.FormatInt(aws.Int64Value(p.VisibilityTimeout), 10)
+	}
+	if p.KmsMasterKeyID != nil {
+		m[v1alpha1.AttributeKmsMasterKeyID] = aws.StringValue(p.KmsMasterKeyID)
+	}
+	if p.KmsDataKeyReusePeriodSeconds != nil {
+		m[v1alpha1.AttributeKmsDataKeyReusePeriodSeconds] = strconv.FormatInt(aws.Int64Value(p.KmsDataKeyReusePeriodSeconds), 10)
+	}
+	if p.ReceiveMessageWaitTimeSeconds != nil {
+		m[v1alpha1.AttributeReceiveMessageWaitTimeSeconds] = strconv.FormatInt(aws.Int64Value(p.ReceiveMessageWaitTimeSeconds), 10)
+	}
+
+	if aws.StringValue(p.RedrivePolicy.DeadLetterQueueARN) != "" {
+		val, err := json.Marshal(p.RedrivePolicy)
+		if err == nil {
+			m[v1alpha1.AttributeRedrivePolicy] = string(val)
+		}
+	}
+	return m
+}
+
+// GenerateQueueTags returns a map of queue tags
+func GenerateQueueTags(tags []v1alpha1.Tag) map[string]string {
+	if len(tags) != 0 {
+		m := map[string]string{}
+		for _, val := range tags {
+			m[val.Key] = val.Value
+		}
+		return m
+	}
+	return nil
+}
+
+// IsErrorNotFound checks if the error returned by AWS API says that the queue being probed doesn't exist
+func IsErrorNotFound(err error) bool {
```

Comment:
I _think_ other instances of this pattern mostly omit `Error` from the name - i.e. `IsNotFound`.